impl/src/main/java/com/okta/sdk/impl/resource/identity/provider/DefaultGoogleIdentityProviderBuilder.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.impl.resource.identity.provider;
+
+import com.okta.sdk.client.Client;
+import com.okta.sdk.resource.identity.provider.GoogleIdentityProviderBuilder;
+import com.okta.sdk.resource.identity.provider.IdentityProvider;
+import com.okta.sdk.resource.identity.provider.IdentityProviderCredentials;
+import com.okta.sdk.resource.identity.provider.IdentityProviderCredentialsClient;
+import com.okta.sdk.resource.identity.provider.OIDCIdentityProviderBuilder;
+import com.okta.sdk.resource.identity.provider.Protocol;
+import com.okta.sdk.resource.identity.provider.ProtocolAlgorithmType;
+import com.okta.sdk.resource.identity.provider.ProtocolAlgorithmTypeSignature;
+import com.okta.sdk.resource.identity.provider.ProtocolAlgorithms;
+import com.okta.sdk.resource.identity.provider.ProtocolEndpoint;
+import com.okta.sdk.resource.identity.provider.ProtocolEndpoints;
+import com.okta.sdk.resource.identity.provider.Provisioning;
+import com.okta.sdk.resource.identity.provider.ProvisioningConditions;
+import com.okta.sdk.resource.identity.provider.ProvisioningDeprovisionedCondition;
+import com.okta.sdk.resource.identity.provider.ProvisioningGroups;
+import com.okta.sdk.resource.identity.provider.ProvisioningSuspendedCondition;
+import com.okta.sdk.resource.policy.IdentityProviderPolicy;
+import com.okta.sdk.resource.policy.PolicyAccountLink;
+import com.okta.sdk.resource.policy.PolicySubject;
+import com.okta.sdk.resource.policy.PolicySubjectMatchType;
+import com.okta.sdk.resource.policy.PolicyUserNameTemplate;
+
+import java.util.List;
+
+public class DefaultGoogleIdentityProviderBuilder implements GoogleIdentityProviderBuilder {
```

Comment:
If we remove the `GoogleIdentityProviderBuilder` interface, and replace it with the OIDC one, this could become an implementation detail, `IdentityProviderBuilder.instance().google()`, or maybe just `IdentityProviderBuilder.google()` and `IdentityProviderBuilder.oidc()` and return the correct implementation.  I really don't know enough about the actual structure of the objects to make a strong recommendation though, it just seems that if we need a Google one, we would also need implementations for LinkedIn and Facebook (i'm hoping the request objects all looks very similar for those),

Reply:
It is a good idea to add builders for all of https://developer.okta.com/docs/reference/api/idps/#add-identity-provider.